README
README file for App::Office::Contacts.

See also: CHANGES, INSTALL.

Warning: WinZip 8.1 and 9.0 both contain an 'accidental' bug which stops
them recognizing POSIX-style directory structures in valid tar files.
You are better off using a reliable tool such as InfoZip:
ftp://ftp.info-zip.org/pub/infozip/

1 Installing from a Unix-like distro
------------------------------------
shell>gunzip CGI-Office-Contacts-1.00.tgz
shell>tar mxvf CGI-Office-Contacts-1.00.tar

On Unix-like systems, assuming you have installed Module::Build V 0.25+:

shell>perl Build.PL
shell>./Build
shell>./Build test
shell>./Build install

On MS Windows-like systems, assuming you have installed Module::Build V 0.25+:

shell>perl Build.PL
shell>perl Build
shell>perl Build test
shell>perl Build install

Alternately, without Module::Build, you do this:

Note: 'make' on MS Windows-like systems may be called 'nmake' or 'dmake'.

shell>perl Makefile.PL
shell>make
shell>make test
shell>su              (for Unix-like systems)
shell>make install
shell>exit            (for Unix-like systems)

On all systems:

Run Contacts.pm through you favourite pod2html translator.

2 Installing from an ActiveState distro
---------------------------------------
shell>unzip CGI-Office-Contacts-1.00.zip
shell>ppm install --location=. CGI-Office-Contacts
shell>del CGI-Office-Contacts-1.00.ppd
shell>del PPM-CGI-Office-Contacts-1.00.tar.gz


CHANGES
Revision history for Perl extension App::Office::Contacts.

1.17  Wed Jun 15  8:15:00 2011
	- Patch Build.PL and Makefile.PL to reduce the version requirement of File::Copy from 2.18 to 2.14,
	  to make it easier for people to install without having to upgrade Perl to get the later version.
	  Thanx to Gabor Szabo for reporting this problem.
	- No other changes.

1.16  Tue Nov 16 15:47:00 2010
	- Switch from FindBin::Real to FindBin (which is in core).
	- Replace /usr/bin/perl with /usr/bin/env perl.
	- Replace common::sense with use strict and use warnings, to get uninit var warnings.
	- Move lib/App-Office-Contacts/lib/App/Office/Contacts/.htoffice.contacts.conf to config/.
	- Change default template path to /dev/shm/html/assets/templates/app/office/contacts.
	- Change name of default template path from tmpl_path to template_path, as part of adopting
	  Text::Xslate.
	- Add scripts/copy.config.pl to copy .htoffice.contacts.conf to ~/.perl/App-Office-Contacts/.
	- Add missing pre-reqs to Build.PL and Makefile.PL.
	- Make Build.PL and Makefile.PL run scripts/copy.config.pl.

1.15  Wed Sep 22 10:00:00 2010
	- Replace sub script_name() with $self -> query -> url(-absolute => 1).
	- Shift some code into a new module, App::Office::CMS::View::Search. This means a view
	  now hasa search.
	- Chop subs generate_cookie(), generate_digest() and validate_post(). See V 1.09 below.

1.14  Fri Jun 25 11:15:00 2010
	- Change all JSON::XS->new->encode(...) to JSON::XS->new->utf8->encode(...).

1.13  Thu Jun 24 14:38:00 2010
	- Fix syntax error.
	- Use 'select count(*) as count' rather than just 'select count(*)' to avoid
	  differences between Postgres and SQLite.

1.12  Wed Jun 23 13:29:00 2010
	- Fix logic error in Create.pm.report_all_tables(). I was getting a list of
	  table names from a file in the distro, the same way I do when populating
	  tables at installation time. But, this file may not be available at run
	  time after installation.
	- No longer ship scripts/schema.sh. I use dbigraph.pl from GraphViz::DBI,
	  modified to use GraphViz::DBI::General (which subclasses GraphViz::DBI).

1.11  Thu Jun  3 17:23:00 2010
	- Fix typos arising after I changed the name of the module
		from CGI::Office::* to App::Office::*. This patch was lost
		when I replaced Debian testing with lenny on my laptop.
	- Ship docs/html/contacts.faq.html, as previously documented.

1.10  Wed May 19 11:11:00 2010
	- Update comments re starman usage in contacts.psgi.
	- Chop mailing list stuff from support.
	- Update version numbers in Build.PL and Makefile.PL.
	- Ensure config code is only called once (App::Office::Contacts::Util::LogConfig).

1.09  Tue Apr 20  8:38:00 2010
	- Comment out the processing which checks for CSRF, since I encountered
		a case where it did not work.

1.08  Fri Apr 16  8:52:00 2010
	- Warning: The organizations and people tables have a new column: upper_name.
		This is due to a defect in SQLite, which does not allow function calls
		when defining an index. Hence the index on people(upper(name) ) has to be
		now written as people(upper_name). You can easily write a little program
		to use alter table, and then populate the new column.
		The search code uses the new column.
	- Change SQLite attribute from unicode to sqlite_unicode.
	- Change the default database driver from Postgres to SQLite,
		to make installation easier (by not requiring DBD::Pg).
		If using another database server, you'll need to edit the 2 lines
		in .htoffice.contacts.conf which refer to SQLite.
	- Fix Makefile.PL to use App::* not CGI::*.
		My apologies for this carelessness.
	- Rework cookies and POST validation, to allow Contacts, Donations and
		Import::vCards to run in parallel.

1.07  Wed Apr  7  8:51:00 2010
	- Update pre-reqs for Test::Pod to 1.41 to avoid Test::Pod's dithering
		about a POD construct I used: L<text|scheme:...>, which makes a test fail.
		See comments for V 1.40 and 1.41 at:
		http://cpansearch.perl.org/src/DWHEELER/Test-Pod-1.42/Changes
	- Update pre-reqs from Test::More V 0 to Test::Simple 0.94.

1.06  Mon Mar 29 14:53:00 2010
	- Create indexes on organizations and people tables, using upper(name),
		to speed up searching. The index names are:
		- organizations: opganizations_upper_name
		- people: people_upper_name.
	- Add parent to pre-reqs in Build.PL and Makefile.PL.

1.05  Tue Mar  2  9:28:00 2010
	- In cgiapp_prerun() protect against XSS and CSRF:
		o Only accept CGI params if the request method is 'POST'.
		o Ensure digest in session matches digest in cookie.
		o http://www.freedom-to-tinker.com/blog/wzeller/popular-websites-vulnerable-cross-site-request-forgery-attacks
	- Change 'use base' to 'use parent'.
	- Remove form_action from config file. See sub script_name.
	- Replace references to FCGI with Plack. This includes no longer
		shipping FCGI-specific files nor patches to Apache's httpd.conf.
	- Ship httpd/cgi-bin/office/contacts.psgi.
	- Adopt Log::Dispatch::Configurator.
		See App::Office::Contacts::Util::LogConfig.
	- Replace Carp::croak with die, assuming calling code uses Try::Tiny.
	- Stop using Time::Elapsed (at table create/populate time).
	- Zap drop_and_create_all_tables() and run() from App::Office::Contacts::Util::Create.
	- In drop.tables.pl and create.tables.pl, change the 'verbose+' option
		definition to 'verbose', since the '+' doesn't make sense.
	- Clean up what is real data and what is fake data.
	- Rename data/email_addresses.txt => data/fake.email_addresses.txt.
	- Rename data/email_people.txt => data/fake.email_people.txt.
	- Rename data/people.txt => data/fake.people.txt.
	- Rename data/phone_numbers.txt => data/fake.phone_numbers.txt.
	- Rename data/phone_people.txt => data/fake.phone_people.txt.
	- Rename data/organizations.txt => data/fake.organizations.txt.
	- Add comments to .htoffice.contacts.conf, while simplifying the
		discussion of the Javascript URL.
	- Change the default URL of the FAQ.
	- Use common::sense instead of strict and warnings.

Makefile.PL
use ExtUtils::MakeMaker;

# See lib/ExtUtils/MakeMaker.pm for details of how to influence
# the contents of the Makefile that is written.

WriteMakefile
(
 ($] ge '5.005') ?
 (
  AUTHOR   => 'Ron Savage (ron@savage.net.au)',
  ABSTRACT => 'A web-based contacts manager',
 ) : (),
 clean =>
 {
	 FILES => 'blib/* Makefile MANIFEST App-Office-Contacts-*'
 },
 dist =>
 {
	 COMPRESS => 'gzip',
	 SUFFIX   => 'gz'
 },
 DISTNAME  => 'App-Office-Contacts',
 NAME      => 'App::Office::Contacts',
# Next line runs copy.config.pl when user runs 'make'.
# Warning: '...' => 0 is OK but => '' or undef are ignored.
 PL_FILES  => {'scripts/copy.config.pl' => 'Copying...'},
 PREREQ_PM =>
 {
	 CGI::Application            => 4.31,
	 CGI::Application::Dispatch  => 2.17,
	 CGI::Session                => 4.42,
	 Config::Tiny                => 2.12,
	 Data::UUID                  => 1.203,
	 Data::Verifier              => 0.31,
	 DateTime                    => 0.55,
	 DBD::SQLite                 => 1.29,
	 DBI                         => 1.611,
	 DBIx::Admin::CreateTable    => 2.04,
	 Digest::SHA                 => 5.48,
	 File::Copy                  => 2.14,
	 File::HomeDir               => 0.97,
	 FindBin                     => 1.49,
	 HTML::Template              => 2.9,
	 JSON::XS                    => 2.29,
	 Log::Dispatch               => 2.26,
	 Log::Dispatch::Configurator => 1.00,
	 Log::Dispatch::DBI          => 0.02,
	 Moose                       => 0.98,
	 Moose::Role                 => 1.21,
	 MooseX::LogDispatch         => 1.2000,
	 namespace::autoclean        => 0.09,
	 parent                      => 0.218,
	 Path::Class                 => 0.23,
	 Perl6::Slurp                => 0.03,
	 Sub::Exporter               => 0.982,
	 Test::Simple                => 0.94,
	 Test::Pod                   => 1.41,
 },
 VERSION_FROM => 'lib/App/Office/Contacts.pm',
);
